sdk/include/xdk/ntbasedef.h

```diff
@@ -787,8 +787,10 @@ extern "C++" { \
  #define RotateRight16 _rotr16
 #endif /* _M_AMD64 */
 
-/* C_ASSERT Definition */
-#define C_ASSERT(expr) extern char (*c_assert(void)) [(expr) ? 1 : -1]
+/* Static assert */
+#ifndef C_ASSERT
+# define C_ASSERT(e) typedef char __C_ASSERT__[(e) ? 1 : -1]
+#endif /* C_ASSERT */
```

Comment:
Note: the intsafe.h has this MSC_VER conditional. Would it be nice to have it there too?
```suggestion
#ifdef _MSC_VER
# define C_ASSERT(e) typedef char __C_ASSERT__[(e)?1:-1]
#else
# define C_ASSERT(e) extern void __C_ASSERT__(int [(e)?1:-1])
#endif
```

Reply:
This `extern` function declaration had a conflict on C++.

Reply:
That's interesting.... Would this mean that e.g. using intsafe.h (from which that extern version comes from) would generate a compile error as well?

Reply:
Seriously, what a mess: see for example those commits:
https://git.reactos.org/?p=reactos.git;a=commit;h=f8e59069e2301e3dd26cce1cadc1d9ac23f932a8
https://git.reactos.org/?p=reactos.git;a=commit;h=f73d776d30f1d4c11f1845d5bfc2a7a7589217b7
> New definition of C_ASSERT and _STATIC_ASSERT using a function prototype. This way we avoid duplicate definition errors like with a typedef and unused function warning like with the extern.

@ThFabba @tkreuzer Can you shed some light on this?

Reply:
Yep if `C_ASSERT`'s of C code and C++ code are mixed in a module that uses `<intsafe.h>`.